workflows/prognostic_c48_run/tests/test_fv3fit_transformer.py

```diff
@@ -117,3 +117,37 @@ def add_one_to_temperature():
     assert "surface_precipitation_rate" not in state
     transform(add_one_to_temperature)()
     assert "surface_precipitation_rate" in state
+
+
+def test_adapter_with_nans(state, tmpdir_factory):
```

Comment:
This is a complex test for the functionality added...mock predictors, temporary files etc. You can make a simpler unit test by refactoring the additions in `apply` to a function `merge_masked_states(left, right)`.

If you worry that function won't be used, then you could add an NaN case to an existing regression test and update the regression data.

Reply:
Good idea.